calicoctl/calico_ctl/profile.py

```diff
@@ -529,3 +533,260 @@ def parse_ports(ports_str):
+        """
+        # Police against empty ids.
+        if not id:
+            return id
+
+        # Check if we have already looked up the network based on the tag.
+        name = self._name_by_id.get(id)
+        if name:
+            return name
+
+        # The tag to network mapping is not found.  use the tag to perform a
+        # Docker network inspect, and check again.
+        self._network_inspect(id)
+        name = self._name_by_id.get(id)
+        if name:
+            return name
+
+        # The mapping is still not found, so we can't map the tag to a Docker
+        # network name.  In that case just return the profile tag.  We also
+        # store the mapping so that when pushing changes back down again we
+        # can ignore rules with these invalid tags - this is necessary when
+        # there are multiple broken rules and you can only delete one at a
+        # time.
+        self._name_by_id[id] = id
+        self._id_by_name[id] = id
+        return id
+
+    def _get_id_from_name(self, name):
+        """
+        Get the profile tag name from the Docker network name.  If the name
+        cannot be mapped to a Docker Network then raise a NoDockerNetwork
+        exception.
+
+        :param name: The Docker network name.
+        :return: The profile tag name.  If the Docker network name is not
+          recognised then raise a NoDockerNetwork exception.
+        """
+        # Police against empty network names
+        if not name:
+            return name
+
+        # Check if we have already looked up the tag based on the network.
+        id = self._id_by_name.get(name)
+        if id:
+            return id
+
+        # The network to tag mapping is not found.  use the net to perform a
+        # Docker network inspect, and check again.
+        self._network_inspect(name)
+        id = self._id_by_name.get(name)
+        if id:
+            return id
+
+        # The network is not found so raise an Exception.
+        raise NoDockerNetwork(name)
+
+    def _network_inspect(self, name):
+        """
+        Perform a Docker network inspect and store the mapping.
+        :param name: The network ID or name.
+        :return: None.
+        """
+        # We need to be using a minimum version of Docker for handling
+        # libnetwork.  Create an appropriate Docker client - if Docker is not
+        # running at an appropriate version then there is nothing more to do
+        # since we can't be running with Calico as a Docker network.
+        if not self.docker_client:
+            try:
+                self.docker_client = docker.Client(version=DOCKER_LIBNETWORK_VERSION,
+                                                   base_url=DOCKER_URL)
+            except docker.errors.APIError:
+                return
+
+        try:
+            network = self.docker_client.inspect_network(name)
+        except docker.errors.APIError:
+            pass
+        else:
+            id = network['Id']
```

Comment:
We are referencing the name and id twice - so we'd need to pull them twice from  network dictionary.  IMHO that sucks more than pulling out the name and ID up front.